react-router/src/Router.md

```diff
@@ -1,115 +1,193 @@
+In this basic setup, we will define three pages: **Home**, **About**, and **Contact**. We’ll also implement navigation using `NavLink`:
 
-4. ** Add the code to `App.jsx` file ** - 
-    - Update the existing code to the following code in ` App.jsx ` file inside the ` src ` folder
 ```jsx
 import React from "react";
-import "./index.css"
+import { BrowserRouter as Router, Route, Routes, NavLink } from "react-router-dom";
+import Home from "./components/Home";
+import About from "./components/About";
+import Contact from "./components/Contact";
 
-export default function App() {
+function App() {
   return (
-    <main>
+    <Router>
       <nav>
-        <ul>
-          <li><a href="/">Home</a></li>
-          <li><a href="/about">About</a></li>
-          <li><a href="/contact">Contact</a></li>
-        </ul>
-        </nav>
-     </main>
+        <NavLink to="/" exact>Home</NavLink>
+        <NavLink to="/about">About</NavLink>
+        <NavLink to="/contact">Contact</NavLink>
+      </nav>
+      <Routes>
+        <Route path="/" exact element={<Home />} />
+        <Route path="/about" element={<About />} />
+        <Route path="/contact" element={<Contact />} />
+      </Routes>
+    </Router>
   );
 }
-// Home Page
-const Home = () => (
-  <Fragment>
-    <h1>Home</h1>
-    <FakeText />
-  </Fragment>
-  );
-// About Page
-const About = () => (
-  <Fragment>
-    <h1>About</h1>
-    <FakeText />
-  </Fragment>
+
+export default App;
+```
+## Advanced Concepts of React Router
+
+### 1. Nested Routes
+
+Nested routes allow you to define sub-routes within a parent route. This is particularly useful for building hierarchical views, such as a user profile section.
+
+```jsx
+// Profile.jsx
+import React from "react";
+import { Outlet, NavLink } from "react-router-dom";
+
+const Profile = () => {
+  return (
+    <div>
+      <h2>User Profile</h2>
+      <nav>
+        <NavLink to="settings">Settings</NavLink>
+        <NavLink to="posts">Posts</NavLink>
+      </nav>
+      <Outlet /> {/* Renders child routes */}
+    </div>
   );
-// Contact Page
-const Contact = () => (
-  <Fragment>
-    <h1>Contact</h1>
-    <FakeText />
-  </Fragment>
+};
+
+// App.jsx
+import Profile from "./components/Profile";
+import Settings from "./components/Settings";
+import Posts from "./components/Posts";
+
+<Router>
+  <Routes>
+    <Route path="profile" element={<Profile />}>
+      <Route path="settings" element={<Settings />} />
+      <Route path="posts" element={<Posts />} />
+    </Route>
+  </Routes>
+</Router>
+```
+### 2. Protected Routes (Authentication)
+
+You can protect certain routes, like user profile pages, by checking if a user is authenticated before rendering a component.
+
+```jsx
+// PrivateRoute.jsx
+import { Navigate } from "react-router-dom";
+
+const PrivateRoute = ({ element: Component, isAuthenticated }) => {
+  return isAuthenticated ? <Component /> : <Navigate to="/login" />;
```

Comment:
Please either mention the code for the `authentication` or tell users that it is not the in-built function , it is made outside and imported here by the user manually for protected routes for better usnderstanding